### PR TITLE
More intuitive legged state indexing

### PR DESCRIPTION
**Pull request by echandler5956f**
See Issue #60

### CODE REVIEW

File: examples/visualization/sol_times.csv
Comment (Akshay5312) [inline examples/visualization/sol_times.csv:1]: We should possibly add this to the gitignore

File: examples/visualization/sol_states.csv
Comment (Akshay5312) [inline examples/visualization/sol_states.csv:1]: We should possibly add this to the gitignore

File: galileo/legged-model/src/LeggedBody.cpp
Comment (Akshay5312) [inline galileo/legged-model/src/LeggedBody.cpp:1]: Not all of this is for legged state indexing right? Maybe a good idea to break it into different PRs?

File: examples/src/go1_test.cpp [hunk 1]
```diff
@@ -1,23 +1,16 @@
...
+#include <random>
 
 int main(int argc, char **argv)
 {
-    double q0[] = {
-        0., 0., 0.339, 0., 0., 0., 1.,
-        0., 0.67, -1.30, 0., 0.67, -1.3, 0., 0.67, -1.3, 0., 0.67, -1.3};
 
-    Eigen::Map<ConfigVector> q0_vec(q0, 19);
+    ConfigVector q0_vec = (ConfigVector(19) << 0., 0., 0.339, 0., 0., 0., 1., 0., 0.67, -1.30, 0., 0.67, -1.3, 0., 0.67, -1.3, 0., 0.67, -1.3).finished();
```
Comment (Akshay5312) [inline examples/src/go1_test.cpp:7]: You're right, this is neater

File: examples/src/go1_test.cpp [hunk 2]
```diff
@@ -1,23 +1,16 @@
...
     std::shared_ptr<LeggedRobotStates> si = robot.si;
 
     casadi::DM X0 = casadi::DM::zeros(si->nx, 1);
-    int j = 0;
-    for (int i = si->nh + si->ndh; i < si->nh + si->ndh + si->nq; ++i)
-    {
-        X0(i) = q0_vec[j];
-        ++j;
-    }
+    X0(casadi::Slice(si->q_index, si->q_index + si->nq), 0) = casadi::DM(std::vector<double>(q0_vec.data(), q0_vec.data() + q0_vec.size()));
```
Comment (Akshay5312) [inline examples/src/go1_test.cpp:13]: love the addition of index positions

**Review by Akshay5312 — APPROVED** [review]

File: examples/visualization/sol_times.csv
Comment (echandler5956f) [reply examples/visualization/sol_times.csv:1]: Yep, love that idea. Keep causing merge conflicts

File: galileo/legged-model/src/LeggedBody.cpp
Comment (echandler5956f) [reply galileo/legged-model/src/LeggedBody.cpp:1]: Yeah, I got a little carried away
